### PR TITLE
fix(feature/improving-code-quality): Improved code quality

### PR DESCRIPTION
**Pull request by saadqu**
- added base service to extend on for data service and other services
- rewritten form validation messages into a seperate component to reuse
- renamed service calls to a more logic name

### CODE REVIEW

File: src/app/form-validation/form-validation.component.html [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<div *ngIf="control.invalid && control.touched" class="alert alert-warning pt-2">
```
Comment (ismail-codinglab) [inline src/app/form-validation/form-validation.component.html]: Remove 'pt-2' it does nothing

- [x] removed 'pt-2'

**Review by ismail-codinglab — COMMENTED** [review]
Please check

File: src/app/form-validation/form-validation.component.html (hunk 1 above)
Comment (saadqu) [reply src/app/form-validation/form-validation.component.html]: Ok

Comment (saadqu) [reply src/app/form-validation/form-validation.component.html]: Done

**Comment by ismail-codinglab** [comment]
Approved

**Pull request merged**